src/App.jsx

```diff
@@ -9,40 +8,100 @@ function App() {
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    const promise = new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []
-          }
-        });
-      }, 2000);
-    });
-
-    promise.then((result) => {
-      setTodoList(result.data.todoList);
+  const fetchData = async () => {
+
+    const options = {
+      method: "GET",      
+      headers: {
+        'Authorization': `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+      },
+    };
+
+    const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+    
+    try {
+      const response = await fetch(url, options);
+    
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const data = await response.json();
+
+      const todos = data.records.map((todo) => ({
+        id: todo.id,
+        title: todo.fields.title
+      }));
+
+      console.log(todos);
+
+      setTodoList(todos);
       setIsLoading(false);
-    });
+      
+    } catch (error) {
+      console.log(error.message);
```

Comment:
If the `try` statement errors out before line 40, `setIsLoading` is never going to update. You should include it in the catch statement or add a `finally` block that can do this should the request pass or fail.